src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1785,6 +1734,128 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
             return CreateConversion(node, operand, conversion, isCast: true, wasCompilerGenerated: wasCompilerGenerated, destination: targetType, diagnostics: diagnostics);
         }
 
+        private void GenerateExplicitConversionErrors(
+            DiagnosticBag diagnostics, 
+            CSharpSyntaxNode syntax, 
+            Conversion conversion, 
+            BoundExpression operand, 
+            TypeSymbol targetType)
+        {
+            // Make sure that errors within the unbound lambda don't get lost.
+            if (operand.Kind == BoundKind.UnboundLambda)
+            {
+                GenerateAnonymousFunctionConversionError(diagnostics, operand.Syntax, (UnboundLambda)operand, targetType);
+                return;
+            }
+
+            if (operand.HasAnyErrors || targetType.IsErrorType())
+            {
+                // an error has already been reported elsewhere
+                return;
+            }
+
+            if (targetType.IsStatic)
+            {
+                // The specification states in the section titled "Referencing Static
```

Comment:
These comments are actually preexisting ones. I have just moved this code into a helper so that I could call it recursively if needed. Yes, sounds like Eric. :-)